### DISCUSSION TITLE
OpenResty无法启动，[crit] ngx_slab_alloc() failed: no memory

### DISCUSSION THREAD

**Discussion by oneryx** [1Panel-dev/1Panel#4983] [Q&A]
### 1Panel 版本

v1.10.6-lts

### 问题描述

全新安装1Panel，安装OpenResty却无法正常启动。
系统有足够内存，可以安装和运行其他软件，比如Stirling-PDF可以正常运行和访问。

系统软件版本如下：
Linux oracle-linux-8 5.4.17-2136.330.7.1.el8uek.aarch64 aarch64 GNU/Linux
1panel version: v1.10.6-lts
mode: stable
1panel/openresty:1.21.4.3-2-1-focal

看论坛中其他人也有一样的问题，可能和aarch64有关？
https://bbs.fit2cloud.com/t/topic/4822

### 重现步骤

无特殊重现步骤，全新安装都会出现，卸载重装也一样。

### 相关日志输出

```shell
2024/05/04 17:30:21 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:30:22 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:30:22 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:30:23 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:30:23 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:30:25 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:30:29 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:30:35 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:30:48 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:31:14 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:32:05 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:33:05 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
2024/05/04 17:34:05 [crit] 1#1: ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory
```

**Comment by wanghe-fit2cloud** [comment]
感谢反馈，我们先在本地环境复现一下上述问题。

**Comment by lg10** [comment]
实际我是用备份来解决的，我安装也出现了这个问题，我在老版本的应用商店里面导出了一个 openresty 的备份，在新版本导入，就可以正常运行了，备份在下面
[openresty_20240511164640.tar.gz](https://github.com/1Panel-dev/1Panel/files/15282495/openresty_20240511164640.tar.gz)

**Reply by yilovef** [reply]
非常感谢，已经解决了，升级后也能正常使用了，万分感谢！！！

**Comment by oneryx** [comment]
@lg10 非常感谢，我导入你这个备份也可以运行

你发现你的镜像是`openresty/openresty:1.21.4.1-7-focal`，openresty官方的，我改为最新的版本`openresty/openresty:1.25.3.1-3-focal`同样可用。

看来是1Panel的版本或者配置的问题。

**Reply by lystormenvoy** [reply]
> @lg10 非常感谢，我导入你这个备份也可以运行
> 
> 你发现你的镜像是`openresty/openresty:1.21.4.1-7-focal`，openresty官方的，我改为最新的版本`openresty/openresty:1.25.3.1-3-focal`同样可用。
> 
> 看来是1Panel的版本或者配置的问题。

可能和lua_shared_dict配置有关

**Comment by lidonglonge** [comment]
Kylin-Server-10-SP2-aarch64-Release-Build09-20210524 也遇到这个问题
ngx_slab_alloc() failed: no memory
nginx: [crit] ngx_slab_alloc() failed: no memory

1panel 版本 v1.10.9-lts

**升级和降级前记得备份**
使用最新的版本openresty/openresty:1.25.3.1-3-focal不行，
通过降级 openresty/openresty:1.21.4.1-7-focal 这个解决
不过创建网站会提示找不到 /usr/local/openresty/1pwaf/data 下的一个文件，把原来的备份替换过去

**Comment by MrZhengI** [comment]
我也遇到了这个问题，但是最新版的不可用，请问有什么解决办法吗，没有备份，之前的东西清掉了。

**Comment by zhengkunwang223** [comment]
修改1panel/apps/openresty/openresty/1pwaf/data/conf/waf.conf文件
将lua_shared_dict waf_locks 100k;
修改为lua_shared_dict waf_locks 200k;
然后重启 Openresty 试试
